hax/treemakers/posrec.py

```diff
@@ -0,0 +1,173 @@
+
+        self.pattern_fitter = PatternFitter(
+            filename=utils.data_file_name(
+                self.pax_config['WaveformSimulator']['s1_patterns_file']),
+            zoom_factor=self.pax_config['WaveformSimulator'].get(
+                's1_patterns_zoom_factor', 1),
+            adjust_to_qe=qes[self.tpc_channels],
+            default_errors=(self.pax_config['DEFAULT']['relative_qe_error'] +
+                            self.pax_config['DEFAULT']['relative_gain_error'])
+        )
+
+        # For s1_aft variable
+        aftmap_filename = utils.data_file_name(hax.config["s1_aft_filename"])
+        self.aft_map = InterpolatingMap(aftmap_filename)
+        self.low_pe_threshold = 10
+
+        # hax.minitrees.Treemaker.__init__(self)
+
+        # load trained NN models
+        nn_model_json = utils.data_file_name(hax.config["neural_network_model"])
+        json_file_nn = open(nn_model_json, 'r')
+        loaded_model_json = json_file_nn.read()
+        json_file_nn.close()
+        loaded_nn_model = model_from_json(loaded_model_json)
+        weights_file = utils.data_file_name(hax.config['neural_network_weights'])
+        loaded_nn_model.load_weights(weights_file)
+        loaded_nn_model.compile(loss='mean_squared_error', optimizer='adam')
+        self.nn_tensorflow = loaded_nn_model
+        self.list_bad_pmts = [1, 2, 12, 26, 34, 62, 65, 79, 86, 88, 102, 118, 130, 134, 135, 139, 148, 150, 152, 162, 178, 183, 190, 198, 206, 213, 214, 234, 239, 244, 27, 73, 91, 137, 167, 203]
+        self.ntop_pmts = 127  # How to get this automatically?
+        tf.keras.backend.clear_session()
+
+    def get_data(self, dataset, event_list=None):
+
+        data, _ = hax.minitrees.load_single_dataset(dataset, ['Corrections',
+                                                              'Fundamentals'])
+        self.x = data.x_3d_nn.values
+        self.y = data.y_3d_nn.values
+        self.z = data.z_3d_nn.values
+
+        self.indices = list(data.event_number.values)
+
+        return hax.minitrees.TreeMaker.get_data(self, dataset, event_list)
+
+    def extract_data(self, event):
+
+        event_data = {
+            "s1_pattern_fit_hax": None,
+            "s1_area_fraction_top_probability_hax": None
+        }
+
+        # We first need the positions. This minitree is only valid when loading
+        # Corrections since you need that to get the corrected positions
+        if not len(event.interactions):
+            return event_data
+        e = event.event_number
+        try:
+            i = self.indices.index(e)
+        except Exception:
+            return event_data
+        interaction = event.interactions[0]
+        s1 = event.peaks[interaction.s1]
+        s2 = event.peaks[interaction.s2]
+
+        # Position reconstruction based on NN from TensorFlow
+        s2apc = np.array(list(s2.area_per_channel))
+        if(len(s2apc)!=self.ntop_pmts):
```

Comment:
@feigaodm @weiyuehuan: in what case does this actually happen? i.e. is it ok to just ignore these events?